scripts/prepare-release-artifact.js

```diff
@@ -0,0 +1,123 @@
+const fs = require('fs');
+const { execSync } = require('child_process');
+const path = require('path');
+const { Octokit } = require('octokit');
+
+const { version } = require('../package.json');
+
+const WORKFLOW_TIMEOUT_MS = 15 * 60 * 1000;
+
+async function sleep(ms) {
+  await new Promise(r => setTimeout(r, ms));
+}
+
+async function fetchWorkflowRun(context) {
+  const { data: workflows } = await context.octokit.rest.actions.listWorkflowRunsForRepo({
+    owner: context.owner,
+    repo: context.repo,
+    branch: process.env.CI_COMMIT_BRANCH ?? 'main',
+  });
+
+  const runs = workflows.workflow_runs;
+
+  const commitSha = process.env.CI_COMMIT_SHA;
+  const run = runs.find(wf => wf.head_sha === commitSha);
+
+  if (run === undefined) {
+    throw new Error(`Workflow not found for commit ${commitSha}`);
+  }
+
+  return run;
+}
+
+async function waitForWorkflowRun(context) {
+  const waitUntil = process.hrtime.bigint() + BigInt(WORKFLOW_TIMEOUT_MS) * 1_000_000n;
```

Comment:
Don't think we need such precision for that. `Date.now()` would be plenty accurate.

Reply:
fixed